packages/app/src/app/pages/common/UserMenu/index.tsx

```diff
@@ -1,44 +1,51 @@
-import * as React from 'react';
 import Relative from '@codesandbox/common/lib/components/Relative';
-import { inject, hooksObserver } from 'app/componentConnectors';
+import React, { FunctionComponent } from 'react';
+
 import { HoverMenu } from 'app/components/HoverMenu';
-import { Menu } from './Menu';
+import { useOvermind } from 'app/overmind';
+
 import { ClickableContainer, ProfileImage } from './elements';
+import { Menu } from './Menu';
+
+export const UserMenu: FunctionComponent = () => {
+  const {
+    actions: {
+      userMenuClosed,
+      modalOpened,
+      signOutClicked,
+      userMenuOpened,
+      files: { gotUploadedFiles },
+    },
+    state: { user, userMenuOpen },
+  } = useOvermind();
+
+  return (
+    <Relative>
+      <ClickableContainer onClick={userMenuOpened}>
+        <ProfileImage
+          alt={user.username}
+          width={30}
+          height={30}
+          src={user.avatarUrl}
+        />
+      </ClickableContainer>
 
-export const UserMenu = inject('store', 'signals')(
-  hooksObserver(
-    ({
-      store: { user, userMenuOpen },
-      signals: {
-        userMenuClosed,
-        modalOpened,
-        signOutClicked,
-        userMenuOpened,
-        files,
-      },
-    }) => (
-      <Relative>
-        <ClickableContainer onClick={userMenuOpened}>
-          <ProfileImage
-            alt={user.username}
-            width={30}
-            height={30}
-            src={user.avatarUrl}
+      {userMenuOpen && (
+        <HoverMenu onClose={() => userMenuClosed()}>
+          <Menu
+            openPreferences={() =>
+              modalOpened({ message: null, modal: 'preferences' })
+            }
+            openStorageManagement={() => gotUploadedFiles()}
```

Comment:
@christianalfoni `gotUploadedFiles` needs an object with a `message` key.
Do I also provide a `null` or do you have a more descriptive one in mind? 🤔

Reply:
The `gotUploadedFiles` is a bit tricky in terms of typing. Cause with `Action` and `Action<string>` we are able to say... "no argument" or "argument of this type". Also we want to refactor all these `{ singleValue }`  payloads to plain `singleValue`, so I think you can just do that here now:

```ts
export const gotUploadedFiles: AsyncAction<{
  message: string;
}> 
```

To:

```ts
export const gotUploadedFiles: AsyncAction<string | void>
```

`void` in Action/AsyncAction means "no argument". I am not sure if it works correctly, cause we evaluate that typing to create the correct signature for the callable action. If it does not work I would just do:

```ts
export const gotUploadedFiles: AsyncAction<string>
```

And pass in `null` as argument. Just make sure any other use of the action passes their values correctly :) 

Does it make sense? :)

Reply:
Makes sense to me. 🙂

If no argument at all, I can just use `Action` or `AsyncAction` right? 🤔

Reply:
Since we're setting `state.currentModalMessage`, the message of `gotUploadedFiles` is required.

https://github.com/codesandbox/codesandbox-client/blob/c22379febc3bbace5539b81baa070bdba50cbdec/packages/app/src/app/overmind/namespaces/files/actions.ts#L220-L237